**Actividad 1 y 2**

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

Preguntas:
1. ¿Cuántas filas y columnas tiene el dataset? R: 38400 filas y 5 columnas
2. ¿Qué tipo de datos contiene la variable Year? R: Contiene el tipo de dato INT
3. ¿Existen valores faltantes? R: Si existen valores faltantes en las columnas Code,Annual change in population y Annual population change (Projected)
4. ¿Cuántas entidades diferentes contiene el dataset? R:Contiene 256 Entidades unicas

1. ¿Quién produjo originalmente los datos? R:Our World in Data 
2. ¿Qué organización los procesa para su utilización en Our World in Data? R: La United Nations
3. ¿Qué representa la variable que analizarás? R: La evolución poblacional 


In [3]:
df.head()


,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
df_year = df[["Year"]]
print(df_year.head())

   Year
0  1951
1  1952
2  1953
3  1954
4  1955


In [5]:
df.shape

(38400, 5)

In [6]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [7]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [8]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [9]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [10]:
df["Entity"].nunique()

256

**Actividad 3**

In [11]:
df_2005 = df[(df["Year"] >= 2005) & (df["Year"] <= 2023) & (df["Entity"] == "Sweden")]
cambio_anual = df_2005["Annual change in population"]

In [12]:
media = cambio_anual.mean()
print("Media", media)

Media 81983.57894736843


In [13]:
mediana = cambio_anual.median()
print("Mediana", mediana)

Mediana 74562.0


In [14]:
varianza = cambio_anual.var()
print("Varianza", varianza)


Varianza 635094171.8128656


In [15]:
desviacion_estandar = cambio_anual.std()
print("Desviación Estándar", desviacion_estandar)


Desviación Estándar 25201.074814635696


In [16]:
minimo = cambio_anual.min()
print("Mínimo", minimo)


Mínimo 35966.0


In [17]:
maximo = cambio_anual.max()
print("Máximo", maximo)

Máximo 134583.0


In [18]:
entidad_min = df.loc[df["Annual change in population"].idxmin(), "Entity"]
entidad_max = df.loc[df["Annual change in population"].idxmax(), "Entity"]
print("Mas bajo:", entidad_min)
print("Mas alto:", entidad_max)

Mas bajo: Ukraine
Mas alto: World


1. ¿Cuál es el valor de la media? R: 81983.57
2. ¿Cuál es el valor de la mediana? R: 74562
3. ¿Cuál es el valor de la desviación estándar? R: 25201.07
4. ¿Qué entidad presenta el valor mínimo y cuál el máximo? R: El mas bajo Ukrania y el mas alto World

**Actividad 4**

In [19]:
import sys
print("Python", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.3


In [20]:
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [21]:
df.loc[10000,["Entity", "Year", "Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

Los datos utilizados en esta práctica provienen de **Our World in Data (OWID)**.


La fuente original es **United Nations, World Population Prospects (2024)**.


**Cita**

UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Inter im Update” [original data].